### DISCUSSION TITLE
Rebuild and test again

### DISCUSSION THREAD

**Discussion by CSmichaelnavara** [geberit/Revit.TestRunner#32] [Ideas]
Hello
This is very useful tool for Revit developers. Thank you very much.

But when I create new feature for Revit, I need to modify my code several times before the tests pass. It is very annoying to restart Revit after each change.

I have forked your repo and implement this feature. It enables you to rebuild both test project and addin project and run tests again and again without restarting Revit. 

If you are interested about this you can look in my repo https://github.com/CSmichaelnavara/Revit.TestRunner and if you want, I can create a pull request to you original version.

Best regards
@csmichaelnavara

**Comment by tobiasfloescher-geberit** [comment]
Hello Michael 

Thank you for sharing your solution. Sounds promising. 
I will definitely have a look soon. 

Best regards, 
Tobias

**Comment by tobiasfloescher-geberit** [comment]
Hello Michael
I have looked at your solution. But what i do not get is the use case. As I understand it right, you are not able to work on a test or feature in your plugin while Revit and TestRunner runs. 

Maybe there are misunderstandings. I show you my process:
- Basic implementation of a feature and a corresponding test 
- Start Revit from VisualStudio (your solution - not TestRunner solution), Debugger is attached
- Start your Test in TestRunner, stop at some breakpoints in test- or feature- code
- Make code changes, Visual Studio's hot reload / edit and continue applies the latest code for the next run
- Coding while Revit runs and the debugger is still attached (ok, sometimes a restart of Revit is necessary)

What is your process. can you give me an example?

Best regards,
Tobias
